source/MarlinProcessors/src/BeamCalClusterReco.cpp

```diff
@@ -1118,6 +1123,9 @@ void BeamCalClusterReco::readSignalHits(LCEvent* evt, LCCollection* colBCal, BCP
                        std::forward_as_tuple(m_BCG->getPadsPerBeamCal(), nullptr));
 
   if (not colBCal or colBCal->getNumberOfElements() == 0) {
+    // Create and add empty collection
+    colBCal = new IMPL::LCCollectionVec(LCIO::CALORIMETERHIT);;
+    evt->addCollection(colBCal, m_hitsOutColName.c_str());
```

Comment:
This should fail, if `colBCal` is already a `CalorimterHit` collection, because the name will be tried to be re-used.

Reply:
I think we have to change to
```
if(not colBCal) {
  return;
}
```
This isn't something we can fix here. 
And then below  (line 1134) check the Type (http://lcio.desy.de/v02-09/doc/doxygen_api/html/classIMPL_1_1LCCollectionVec.html#ae5664ea9c2ee52d6b1edba8b89e9aa43)  of the collection instead of dynamic_casting and if it is SIMCALORIMETERHIT create the CalorimterHit Collection